src/Jobs/UpdateSyncableProductsCount.php

```diff
@@ -0,0 +1,115 @@
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\SyncableProductsBatchedActionSchedulerJobTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\TransientsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductRepository;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class UpdateSyncableProductsCount
+ *
+ * Get the number of syncable products (i.e. product ready to be synced to Google Merchant Center) and update it in the DB.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Jobs
+ * @since x.x.x
+ */
+class UpdateSyncableProductsCount extends AbstractBatchedActionSchedulerJob {
+	use SyncableProductsBatchedActionSchedulerJobTrait;
+
+	/**
+	 * @var ProductRepository
+	 */
+	protected $product_repository;
+
+	/**
+	 * @var TransientsInterface
+	 */
+	protected $transients;
+
+	/**
+	 * @var int
+	 */
+	private $count;
+
+	/**
+	 * UpdateSyncableProductsCount constructor.
+	 *
+	 * @param ActionSchedulerInterface  $action_scheduler
+	 * @param ActionSchedulerJobMonitor $monitor
+	 * @param ProductRepository         $product_repository
+	 * @param TransientsInterface       $transients
+	 */
+	public function __construct( ActionSchedulerInterface $action_scheduler, ActionSchedulerJobMonitor $monitor, ProductRepository $product_repository, TransientsInterface $transients ) {
+		parent::__construct( $action_scheduler, $monitor );
+		$this->product_repository = $product_repository;
+		$this->transients         = $transients;
+		$this->count              = 0;
+	}
+
+	/**
+	 * Get the count instance variable.
+	 *
+	 * @return int
+	 */
+	public function get_count(): int {
+		return $this->count;
+	}
+
+	/**
+	 * Set the count instance variable.
+	 *
+	 * @param int $count The number that will be set to $this->count.
+	 */
+	public function set_count( $count ) {
+		$this->count = $count;
+	}
+
+	/**
+	 * Get the name of the job.
+	 *
+	 * @return string
+	 */
+	public function get_name(): string {
+		return 'update_syncable_products_count';
+	}
+
+	/**
+	 * Get job batch size.
+	 *
+	 * @return int
+	 */
+	protected function get_batch_size(): int {
+		/**
+		 * Filters the batch size for the job.
+		 *
+		 * @param string Job's name
+		 */
+		return apply_filters( 'woocommerce_gla_batched_job_size', 500, $this->get_name() );
+	}
+
+	/**
+	 * Process batch items.
+	 *
+	 * @param int[] $items A single batch of WooCommerce product IDs from the get_batch() method.
+	 */
+	protected function process_items( array $items ) {
+		$this->set_count( $this->get_count() + count( $items ) );
+	}
+
+	/**
+	 * Called when the job is completed.
+	 *
+	 * @param int $final_batch_number The final batch number when the job was completed.
+	 *                                If equal to 1 then no items were processed by the job.
+	 */
+	protected function handle_complete( int $final_batch_number ) {
+		$this->transients->set( TransientsInterface::SYNCABLE_PRODUCTS_COUNT, $this->get_count(), HOUR_IN_SECONDS );
```

Comment:
Is an hour long enough? If we aren't going to recount until we restart onboarding, then why does it need an expiry time at all? Should it be moved to an option?

Reply:
Thanks for the suggestion. I think using an option here is more appropriate than using a transient, as you said we would only recalculate when we restart onboarding.

Reply:
Updated in 25f3ee3.